### PR TITLE
Add code for analyzing docking results

### PR DESCRIPTION
**Pull request by apayne97**
## Todos
  - [x] generalize the `combine_files` function to combine generated SDFs
  - [x] add set of results_csv column names to analysis.py DockingResults class
  - [x] change default arguments in the DockingResults class to be immutable types (and change all uses of the class)
  - [x] add script to generate a "fragalysis-like" directory from a csv file of best results

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by apayne97** [comment]
Ok I did a terrible thing with rebasing, clearly I need to learn how git works again.

File: asapdiscovery-docking/asapdiscovery/docking/scripts/fauxalysis_from_best_results_csv.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    if not len(sdf_paths) == len(structure_paths):
+        raise ValueError(
+            f"Loaded {len(sdf_paths)} sdf paths and {len(structure_paths)} structure paths, looks like some are missing!"
+        )
+
+    # Copy sdfs and protein pdbs to output dir
+    # Keep track of which sdf files belong to which structure source
+    sdfs_per_structure = {structure_name: [] for structure_name in structure_names}
+    for sdf_path, structure_path, dir_name, structure_name in zip(
```
Comment (hmacdope) [inline asapdiscovery-docking/asapdiscovery/docking/scripts/fauxalysis_from_best_results_csv.py:69]: Use `df.iterrows()` to iterate over dataframe directly.

**Review by hmacdope — COMMENTED** [review]
Merge when ready.

**Review by hmacdope — APPROVED** [review]
Approving for CI, merge when ready.

**Pull request merged**